File: backend/src/main/scala/microservices/operations/overall/objects/ManagerTasksPlannerRequest.scala
// 本文件定义 `ListManagerTasksPlanner` 对应的请求对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerTasksPlannerRequest(managerId: String, managerType: String, taskStatus: Option[String], taskResourceType: Option[String])
object ManagerTasksPlannerRequest:
  given sourceEncoder: Encoder[ManagerTasksPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ManagerTasksPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/overall/objects/ManagerTasksPlannerRequest.ts
// 本文件定义 `ListManagerTasksPlanner` 对应的请求对象。

export type ManagerTasksPlannerRequest = {
  managerId: string
  managerType: string
  taskStatus?: string | null
  taskResourceType?: string | null
}

export const managerTasksPlannerRequestFromJson = (json: string): ManagerTasksPlannerRequest =>
  JSON.parse(json) as ManagerTasksPlannerRequest

export const managerTasksPlannerRequestToJson = (value: ManagerTasksPlannerRequest): string =>
  JSON.stringify(value)
